pkg/local_object_storage/metabase/delete.go

```diff
@@ -141,25 +130,15 @@ func (db *DB) deleteGroup(tx *bbolt.Tx, addrs []oid.Address, sizes []uint64) (ui
 		}
 	}
 
-	for _, refNum := range refCounter {
-		if refNum.cur == refNum.all {
-			err := db.deleteObject(tx, refNum.obj, true)
-			if err != nil {
-				return rawDeleted, availableDeleted, err // maybe log and continue?
-			}
-		}
-	}
-
 	return rawDeleted, availableDeleted, nil
 }
 
-// delete removes object indexes from the metabase. Counts the references
-// of the object that is being removed.
+// delete removes object indexes from the metabase.
```

Comment:
So, you're relying on engine to drop every object from the metabase now, right?

Reply:
yes, that is the point. i started to think that "complex" logic should be done on the engine level and that is why the engine exists. If a metabase is trying to handle parts/parents that it may lose (resync or it just never saw any part with parent header), there are always some problems. that is what i want to discuss and open another issue if any needed